python/sglang/srt/model_executor/hpu_graph_runner.py

```diff
@@ -0,0 +1,372 @@
+# Copyright 2023-2024 SGLang Team
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Run the model with hpu graph."""
+
+from __future__ import annotations
+
+import logging
+import math
+import os
+import time
+from collections import namedtuple
+from contextlib import contextmanager
+from typing import TYPE_CHECKING
+
+import torch
+import tqdm
+
+from sglang.srt.layers.logits_processor import LogitsProcessorOutput
+from sglang.srt.model_executor.forward_batch_info import (
+    CaptureHiddenMode,
+    ForwardBatch,
+    ForwardMode,
+)
+from sglang.srt.utils import is_hpu
+
+_is_hpu = is_hpu()
+if _is_hpu:
+
+    os.environ["PT_HPU_ENABLE_LAZY_COLLECTIVES"] = "true"
```

Comment:
This is lazy specific, so please change accordingly, not needed for compile mode.

Reply:
IMO no need to set here explicitly.